modin/core/dataframe/pandas/partitioning/partition.py

```diff
@@ -171,16 +171,21 @@ def _iloc(df, row_labels, col_labels):  # noqa: RT01, PR01
         """Perform `iloc` on dataframes wrapped in partitions (helper function)."""
         return df.iloc[row_labels, col_labels]
 
-    def mask(self, row_labels, col_labels):
+    def mask(self, row_labels, col_labels, order_broken=False):
         """
         Lazily create a mask that extracts the indices provided.
 
+        By default, assumes that the order of the requested roes/cols is
+        the same as the order in which they are stored.
+
         Parameters
         ----------
         row_labels : list-like, slice or label
             The row labels for the rows to extract.
         col_labels : list-like, slice or label
             The column labels for the columns to extract.
+        order_broken : bool, default: False
+            If the value is True, then some optimizations are not applicable.
```

Comment:
let's elaborate on what optimizations are not applicable :)
after reading the docstring it doesn't clear at all what this parameter is doing and when you should pass it `True` or `False`

BTW, maybe some other name such as `ignore_locators_order` or something will be better?